src/components/course/data/hooks.jsx

```diff
@@ -420,3 +421,28 @@ export const useTrackSearchConversionClickHandler = ({ href, eventName }) => {
 
   return handleClick;
 };
+
+/**
+ * Returns a function to be used as a click handler that emits an optimizely enrollment click event.
+ *
+ * @param {object} args
+ *
+ * @returns Click handler function for clicks on enrollment buttons.
+ */
+export const useOptimizelyEnrollmentClickHandler = () => {
+  const {
+    state: {
+      activeCourseRun: { key: courseKey },
+    },
+  } = useContext(CourseContext);
+  const handleClick = useCallback(
+    (e) => {
+      pushEnrollmentClickEvent({
```

Comment:
Through calling this function, I think we will be making an asynchronous call to an Optimizely API. Because it seems this click handler is being used alongside a native hyperlink (`a`), we will need to override the click behavior similar to what was implemented in `useTrackSearchConversionClickHandler`, i.e.:

```
// if tracking is on a link with an external href destination, we must intentionally delay the default click
// behavior to allow enough time for the async analytics event call to resolve.
```

This is also how Segment.js's `trackLink` function works. If the Optimizely API call happens to take longer than the browser's ability to redirect to the next URL due to a slow network, the event won't be sent. This intentional slight delay gives us more of a chance to capture the event in those cases without being perceivable in the user experience.

Reply:
Good call!

Reply:
@iloveagent57 Perhaps? Though, is it creating the timeouts in parallel vs. one after the other though? If both timeouts are created roughly at the same time, they should still both resolve within the same amount of time, rather than one being blocked by another.